src/DDTrace/Data/Span.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace DDTrace\Data;
+
+use DDTrace\Time;
+use DDTrace\Data\SpanContext as SpanContextData;
+use DDTrace\Contracts\Span as SpanInterface;
+use DDTrace\Integrations\Integration;
+
+abstract class Span implements SpanInterface
+{
+    /**
+     * Operation Name is the name of the operation being measured. Some examples
+     * might be "http.handler", "fileserver.upload" or "video.decompress".
+     * Name should be set on every span.
+     *
+     * @var string
+     */
+    public $operationName;
+
+    /**
+     * @var SpanContextData
+     */
+    public $context;
+
+    /**
+     * Resource is a query to a service. A web application might use
+     * resources like "/user/{user_id}". A sql database might use resources
+     * like "select * from user where id = ?".
+     *
+     * You can track thousands of resources (not millions or billions) so
+     * prefer normalized resources like "/user/{id}" to "/user/123".
+     *
+     * Resources should only be set on an app's top level spans.
+     *
+     * @var string
+     */
+    public $resource;
+
+    /**
+     * Service is the name of the process doing a particular job. Some
+     * examples might be "user-database" or "datadog-web-app". Services
+     * will be inherited from parents, so only set this in your app's
+     * top level span.
```

Comment:
Again I am not sure that this is the spirit of the service. Why should a PDO span inherit the service name `payments-api` instead of having the service `pdo`?

Reply:
👍